distributed/client.py

```diff
@@ -2635,23 +2636,49 @@ def _graph_to_futures(
 
             # Create futures before sending graph (helps avoid contention)
             futures = {key: Future(key, self, inform=False) for key in keyset}
-            self._send_to_scheduler(
-                {
-                    "op": "update-graph",
-                    "tasks": valmap(dumps_task, dsk),
-                    "dependencies": dependencies,
-                    "keys": list(map(tokey, keys)),
-                    "restrictions": restrictions or {},
-                    "loose_restrictions": loose_restrictions,
-                    "priority": priority,
-                    "user_priority": user_priority,
-                    "resources": resources,
-                    "submitting_task": getattr(thread_state, "key", None),
-                    "retries": retries,
-                    "fifo_timeout": fifo_timeout,
-                    "actors": actors,
-                }
-            )
+
+            # Just for testing "update-graph-hlg"
+            if not isinstance(dsk, HighLevelGraph):
+                dsk = HighLevelGraph.from_collections(id(dsk), dsk, dependencies=())
+
+            if isinstance(dsk, HighLevelGraph):
+                dsk = dsk.map_tasks(dumps_task)
+
+                self._send_to_scheduler(
+                    {
+                        "op": "update-graph-hlg",
+                        "hlg": dsk,
```

Comment:
I'm surprised here.  Will msgpack respect the `__reduce__` method?

Reply:
It doesn't natively but https://github.com/dask/distributed/pull/4140/files#diff-7a335df475cb573124e2aedec25f055aR118 implement the support.